crates/matrix-sdk-ui/src/timeline/tests/edit.rs

```diff
@@ -163,3 +168,108 @@ async fn test_aggregated_sanitized() {
+            "format": "org.matrix.custom.html",
+            "body": "**original** message",
+            "msgtype": "m.text"
+        },
+        "event_id": &original_event_id,
+        "origin_server_ts": timeline.event_builder.next_server_ts(),
+        "sender": *ALICE,
+        "type": "m.room.message",
+    });
+
+    let mut event: SyncTimelineEvent = raw_event.into();
+
+    let mut encryption_info = EncryptionInfo {
+        sender: (*ALICE).into(),
+        sender_device: None,
+        algorithm_info: AlgorithmInfo::MegolmV1AesSha2 {
+            curve25519_key: "123".to_owned(),
+            sender_claimed_keys: BTreeMap::new(),
+        },
+        verification_state: VerificationState::Verified,
+    };
+
+    event.encryption_info = Some(encryption_info.clone());
+
+    timeline.handle_live_event(event).await;
+
+    let items = timeline.inner.items().await;
+    let first_event = items[1].as_event().unwrap();
+
+    assert_eq!(
+        first_event.encryption_info.as_ref().unwrap().verification_state,
+        VerificationState::Verified
+    );
+
+    assert_let!(TimelineItemContent::Message(message) = first_event.content());
+    assert_let!(MessageType::Text(text) = message.msgtype());
+    assert_eq!(text.body, "**original** message");
+
+    let raw_event = sync_timeline_event!({
+        "content": {
+            "formatted_body": "<strong>original</strong> message",
+            "format": "org.matrix.custom.html",
+            "body": "**original** message",
+            "msgtype": "m.text"
+        },
+        "event_id": &original_event_id,
+        "origin_server_ts": timeline.event_builder.next_server_ts(),
+        "sender": *ALICE,
+        "type": "m.room.message",
+        "unsigned": {
+            "m.relations": {
+                "m.replace": {
+                    "content": {
+                        "formatted_body": "* <edited/> <strong>better</strong> message",
+                        "format": "org.matrix.custom.html",
+                        "body": "* !!edited!! **better** message",
+                        "m.new_content": {
+                            "formatted_body": "<edited/> <strong>better</strong> message",
+                            "format": "org.matrix.custom.html",
+                            "body": "!!edited!! **better** message",
+                            "msgtype": "m.text"
+                        },
+                        "m.relates_to": {
+                            "event_id": original_event_id,
+                            "rel_type": "m.replace"
+                        },
+                        "msgtype": "m.text"
+                    },
+                    "event_id": EventId::new(server_name!("dummy.server")),
+                    "origin_server_ts": timeline.event_builder.next_server_ts(),
+                    "sender": *ALICE,
+                    "type": "m.room.message",
+                }
+            }
+        }
+    });
```

Comment:
Any chance you can use the fancy `EventFactory` instead? There's a team-wide effort to get rid of `sync_timeline_event` so as to get to more readable code (https://github.com/matrix-org/matrix-rust-sdk/issues/3716). Looks like the way to create edits I've added in #3854 would be sufficient to create the edit event.

Reply:
Whoa, that's sooo much nicer in comparison. I can go ahead and rebase this on top of your 3845 and switch to it if you agree.